mlflow/store/artifact/artifact_repo.py

```diff
@@ -319,6 +319,9 @@ def download_trace_data(self) -> Dict[str, Any]:
             - `MlflowTraceDataNotFound`: The trace data is not found.
             - `MlflowTraceDataCorrupted`: The trace data is corrupted.
         """
+        if not any(a.path == TRACE_DATA_FILE_NAME for a in self.list_artifacts()):
+            raise MlflowTraceDataNotFound(artifact_path=TRACE_DATA_FILE_NAME)
```

Comment:
I don't think this is ideal. The ideal approach is handle this in each artifact repository.

Reply:
`list_artifacts` is not a terrible idea but a small overhead could be amplified when `search_traces` calls `download_trace_data` a lot.

Reply:
offline discussion: catch `Exception` and throw `MlflowTraceDataNotFound` instead. This might not be correct (e.g. `traces.json` exist but download failed due to a transient network issue).